dll/win32/shell32/CShellLink.cpp

```diff
@@ -2802,6 +2802,29 @@ BOOL CShellLink::OnInitDialog(HWND hwndDlg, HWND hwndFocus, LPARAM lParam)
     if (m_sDescription)
         SetDlgItemTextW(hwndDlg, IDC_SHORTCUT_COMMENT_EDIT, m_sDescription);
 
+    /* Hot key */
+    SendDlgItemMessageW(hwndDlg, IDC_SHORTCUT_KEY_HOTKEY, HKM_SETHOTKEY, m_Header.wHotKey, 0);
+
+    /* Run */
+    const WORD runstrings[] = { IDS_SHORTCUT_RUN_NORMAL, IDS_SHORTCUT_RUN_MIN, IDS_SHORTCUT_RUN_MAX };
+    const BYTE runshowcmd[] = { SW_SHOWNORMAL, SW_SHOWMINNOACTIVE, SW_SHOWMAXIMIZED };
+    HWND hRun = GetDlgItem(hwndDlg, IDC_SHORTCUT_RUN_COMBO);
+    for (UINT i = 0; i < _countof(runstrings); ++i)
+    {
+        WCHAR buf[MAX_PATH];
+        if (!LoadString(shell32_hInstance, runstrings[i], buf, _countof(buf)))
+            break;
+
+        int idx = SendMessage(hRun, CB_ADDSTRING, 0, (LPARAM)buf);
+        if (idx < 0)
+            continue;
+        SendMessage(hRun, CB_SETITEMDATA, idx, runshowcmd[i]);
+        if (!i || m_Header.nShowCommand == runshowcmd[i])
```

Comment:
Is there ever a case where `m_Header.nShowCommand == runshowcmd[i]` does not yield a selection?

Reply:
If the .lnk file is corrupted, that is why it sets it to normal just in case (the `!i` part). Windows only allows these 3 values in a shortcut.

The [MS specification ](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-shllink/c3376b21-0931-45e4-b2fc-a48ac0e60d15) spells this out: 

> This value SHOULD be one of the following.
> ...
> All other values MUST be treated as SW_SHOWNORMAL.